models/inventory/StockManager.ts

```diff
@@ -137,32 +137,58 @@ export class StockManager {
         details.item,
         details.fromLocation,
         undefined,
-        date
+        date,
+        undefined
       )) ?? 0;
 
+
     const formattedDate = this.fyo.format(details.date, 'Datetime');
 
     if (this.isCancelled) {
       quantityBefore += details.quantity;
     }
 
+    // batch-wise stock validation
+    let itemsInBatch =
+      (await this.fyo.db.getStockQuantity(
+        details.item,
+        details.fromLocation,
+        undefined,
+        date,
+        details.batchNumber
+      )) ?? 0;
+
+    
+    let ifItemHasBatchNumber= await this.fyo.getValue('Item', details.item, 'hasBatchNumber');
```

Comment:
Check batch wise stock only if item has batch enabled. Need to rework a bit on this file.